app/models/post_timing.rb

```diff
@@ -40,6 +40,10 @@ def self.record_new_timing(args)
     # still happen, if it happens we just don't care, its an invalid record anyway
     return if row_count == 0
     Post.where(['topic_id = :topic_id and post_number = :post_number', args]).update_all 'reads = reads + 1'
+
+    topic = Topic.select(:archetype).find_by(id: args[:topic_id])
+    return if topic&.archetype == Archetype.private_message
```

Comment:
🏌️⛳ 
```rb
    return if Topic.find_by(id: args[:topic_id], archetype: Archetype.private_message)
```

(I don't think it's much better so I'm not even marking it as a suggestion 😄)

Reply:
I ended up using `exists?` instead:

```ruby
Topic.exists?(id: args[:topic_id], archetype: Archetype.private_message)
```

Thanks for the suggestion. 😄